### DISCUSSION TITLE
Configuration schema

### DISCUSSION THREAD

**Discussion by b4nst** [vectordotdev/vector#19423] [Q&A] [answered]
Hey, is there a schema of the Vector config available somewhere?

We're using CUE to generate our vector config, and we would like to add typing to it.
Something like a Json schema would be useful.

**Comment by jszwedko** [comment, marked as answer]
Hey! There is actually a way to generate a JSON schema describing the config via `vector generate-schema`. Just a note that this is experimental, though, and we don't make any backwards compatibility guarantees.

**Reply by b4nst** [reply]
That's already a big win, thanks!
